### ISSUE TITLE
Cannot progressively reveal sub-sections that follow a sub-section that has no exercises

### ISSUE THREAD

**Issue by garrettgman** [rstudio/tutor#25]
Progressive reveal seems to rely on exercise completion to trigger reveals. As a result, you cannot reveal the next sub-section if the current sub-section does not contain an exercise. Related to #26.

To see this in action, launch the tutorial below and then compare the behavior of "First Section" to the behavior of "Second Section".

````
---
title: "Demo Tutorial"
output: 
  tutor::tutorial:
    progressive: true
runtime: shiny_prerendered
---

```{r setup, include=FALSE}
library(tutor)
checker <- function(label, user_code, check_code, envir_result, evaluate_result, ...) {
  list(message = "Good Job!", correct = TRUE, location = "append")
}
tutor_options(exercise.checker = checker)
```

## Welcome

This is the Welcome section. It has an exercise to ensure that the TOC appears. Click "Submit Code" to move on.

```{r one, exercise = TRUE}

```

```{r one-check}
"foo"
```

## First task

### Visible sub-section

This sub-section is visible. If it had an exercise to complete, you could move on to the next sub-section, which is hidden. Notice that there is no "Next Topic" button.

### Hidden sub-section

This sub-section is hidden. You cannot access it because there is no exercise to complete in the previous section.

## Second Task

### Another Visible sub-section

This is the Second Task section. It has an exercise to complete, so you can move on to the hidden sub-section.

This is the Welcome section. It has an exercise to ensure that the TOC appears. Click "Submit Code" to move on.

```{r two, exercise = TRUE}

```

```{r two-check}
"foo"
```

### Another Hidden sub-section

This section is hidden, but you can reveal it by completing the exercise above. Notice that there is no "Next Topic" button.
````

**Comment by robbyshaver** [comment]
The issue here is that allow_skip wasn't enabled in the RMD - it must be on in order to skip over a section without an exercise to the next section. However, there is a bug if a document has more than one section without an exercise - I've logged issue #33 to track that problem.

**Comment by robbyshaver** [comment]
In progressive reveal any section without an exercise should have a skip/go on button regardless of whether or not skip is enabled.

**Comment by garrettgman** [comment]
Thanks for reopening this @robbyshaver. Possibly tied to #32 and #33.

**Comment by robbyshaver** [comment]
I never closed this one - it's #26 I closed because I think it's covered by other ones.

**Comment by robbyshaver** [comment]
Allow skip does not have to be enabled to skip a section with no exercises.
